### PR TITLE
Ensure that necessary folder exists

### LINKED ISSUE #35: The hardcoded path may not exist, it should ensure it
# 🐛 Summary #

- /var/log/clamav may not exist
- /var/log/clamav/lsatscan.log may be world readable

## To reproduce ##

1. install in fresh Operating System
2. run clamscan

PR coming

### PR DESCRIPTION
**Pull request by ClementJ35**
## 🗣 Description ##

Ensure that folder exists as described in #35

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -21,6 +21,18 @@
     dest: /etc/cron.weekly/virus_scan
     mode: 0755
 
+- name: Ensure that log folder exist
+  ansible.builtin.file:
+    path: /var/log/clamav
+    state: directory
+    mode: '0755'
```
Comment (jsf9k) [inline tasks/main.yml]: Remove unnecessary quotes:
```suggestion
    mode: 0755
```

File: tasks/main.yml [hunk 2]
```diff
@@ -21,6 +21,18 @@
...
+
+- name: Ensure log file will be world-readable
+  ansible.builtin.file:
+    path: /var/log/clamav/lastscan.log
+    state: file
+    mode: '0644'
```
Comment (jsf9k) [inline tasks/main.yml]: Remove unnecessary quotes:
```suggestion
    mode: 0644
```

File: tasks/main.yml [hunk 3]
```diff
@@ -21,6 +21,18 @@
     dest: /etc/cron.weekly/virus_scan
     mode: 0755
 
+- name: Ensure that log folder exist
```
Comment (jsf9k) [inline tasks/main.yml]: Fix typo:
```suggestion
- name: Ensure that log folder exists
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
Requesting a few very minor changes.  Also, please address the failing tests.

**Comment by ClementJ35** [comment]
As stated here : https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html, the only way to create an empty file that doesn't exist is touch. Using copy or template would result in blanking the log if it exists.
But touch seems to break idempotence autobuild test.
What next ?

**Comment by jsf9k** [comment]
> As stated here : https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html, the only way to create an empty file that doesn't exist is touch. Using copy or template would result in blanking the log if it exists. But touch seems to break idempotence autobuild test. What next ?

You can use [`ansible.builtin.stat`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html) to determine whether you need to actually do the `touch` or not.  Let me know if you can't get it to work and I will make the change for you.

**Comment by jsf9k** [comment]
@ClementJ35, thanks for the improvements to this Ansible role!

In the future, when accepting suggestions (like you did in commit 99749ef, for example) please change the default commit message to something more specific to the modifications in the commit.  For example, "Remove unnecessary quotes" would have been a good commit message for 99749ef.

Thanks again for the contribution...we appreciate it!

**Comment by mcdonnnj** [comment]
@ClementJ35 Just a thought but can we use `access_time: preserve` and `modification_time: preserve` to clear idempotence while using `state: touch`?

**Comment by jsf9k** [comment]
> @ClementJ35 Just a thought but can we use `access_time: preserve` and `modification_time: preserve` to clear idempotence while using `state: touch`?

This is worth a try, @ClementJ35.  This is a simpler and cleaner solution than what I suggested...if it works.

**Comment by jsf9k** [comment]
@ClementJ35 - I'm sorry for my name being attached to commits 5f9ba14 and 57a0a2c.  I rebased and force-pushed in order to make minor changes to the commit messages to better align with our processes, and `git` did that.  It wasn't my intent.

Thanks for another great pull request on this repo!

**Review by jsf9k — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
Seems reasonable enough and LGTM ✔

**Comment by ClementJ35** [comment]
Hi @jsf9k @mcdonnnj,
Those changes are fine to me. I just didn't thought about time preservation for idempotence.
@jsf9k, don't worry about my name, I don't care. I just want that role to do better.
Thanks for your responsiveness.

**Review by dav3r — APPROVED** [review]
Love it!  Nice improvement.  💎

**Pull request merged**